SPECS/pam_wrapper/pam_wrapper.spec

```diff
@@ -0,0 +1,208 @@
+Requires:       pam_wrapper = %{version}-%{release}
+
+%description -n libpamtest
+If you plan to test a PAM module you can use this library, which simplifies
+testing of modules.
+
+%package -n libpamtest-devel
+Summary:        A tool to test PAM applications and PAM modules
+Requires:       libpamtest = %{version}-%{release}
+Requires:       pam_wrapper = %{version}-%{release}
+Recommends:     cmake
+Recommends:     pkg-config
+
+%description -n libpamtest-devel
+If you plan to develop tests for a PAM module you can use this library,
+which simplifies testing of modules. This sub package includes the header
+files for libpamtest.
+
+%package -n libpamtest-doc
+Summary:        The libpamtest API documentation
+
+%description -n libpamtest-doc
+Documentation for libpamtest development.
+
+%package -n python3-libpamtest
+Summary:        A python wrapper for libpamtest
+Requires:       libpamtest = %{version}-%{release}
+Requires:       pam_wrapper = %{version}-%{release}
+
+%description -n python3-libpamtest
+If you plan to develop python tests for a PAM module you can use this
+library, which simplifies testing of modules. This subpackage includes
+the header files for libpamtest
+
+%prep
+%autosetup -S git
+
+
+%build
+if test ! -e "obj"; then
+  mkdir obj
+fi
+pushd obj
+%cmake \
+  -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+  -DUNIT_TESTING=ON \
+  %{_builddir}/%{name}-%{version}
+
+make %{?_smp_mflags} VERBOSE=1
+make doc VERBOSE=1
+popd
+
+
+%install
+pushd obj
+make DESTDIR=%{buildroot} install
+popd
+
+%post -p /sbin/ldconfig
+%postun -p /sbin/ldconfig
+%post -p /sbin/ldconfig -n libpamtest
+%postun -p /sbin/ldconfig -n libpamtest
+
+%check
+pushd obj
+ctest --output-on-failure
+popd
+
+%files
+%{_libdir}/libpam_wrapper.so*
+%{_libdir}/pkgconfig/pam_wrapper.pc
+%dir %{_libdir}/cmake/pam_wrapper
+%{_libdir}/cmake/pam_wrapper/pam_wrapper-config-version.cmake
+%{_libdir}/cmake/pam_wrapper/pam_wrapper-config.cmake
+%{_libdir}/pam_wrapper/pam_chatty.so
+%{_libdir}/pam_wrapper/pam_matrix.so
+%{_libdir}/pam_wrapper/pam_get_items.so
+%{_libdir}/pam_wrapper/pam_set_items.so
+%{_mandir}/man1/pam_wrapper.1*
+%{_mandir}/man8/pam_chatty.8*
+%{_mandir}/man8/pam_matrix.8*
+%{_mandir}/man8/pam_get_items.8*
+%{_mandir}/man8/pam_set_items.8*
+
+%files -n libpamtest
+%{_libdir}/libpamtest.so.*
+
+%files -n libpamtest-devel
+%{_libdir}/libpamtest.so
+%{_libdir}/pkgconfig/libpamtest.pc
+%dir %{_libdir}/cmake/pamtest
+%{_libdir}/cmake/pamtest/pamtest-config-relwithdebinfo.cmake
+%{_libdir}/cmake/pamtest/pamtest-config-version.cmake
+%{_libdir}/cmake/pamtest/pamtest-config.cmake
+%{_includedir}/libpamtest.h
+
+%files -n libpamtest-doc
+%doc obj/doc/html
+
+%files -n python3-libpamtest
+%{_lib}/python3.7/site-packages/pypamtest.so
+
+%changelog
+* Mon Jul 26 2021 Shane Guan <shaneguan@microsoft.com> - 1.1.3-4
+- Replace ldconfig_scriptlets with the expanded form
+- Replace gpgv2 with gpg
+- on our build system, python3 packages are located in %{_lib}/python3.7
```

Comment:
>on our build system, python3 packages are located in %{_lib}/python3.7 [](http://example.com/codeflow?start=2&length=70)

Not sure what this line is about. Seems like added by mistake. Also, not sure if that's true: most places use `%{_lib}/python3.7`. I think only `WALinuxAgent.spec` is an exception. We should probably fix that...:P

Reply:
This is referring to how in the original parity version the spec had this line in %files
```
%files -n python3-libpamtest
%{python3_sitearch}/pypamtest.so
```
I think this made the spec not build since the macro wasn't defined, so I replaced it to the following 
```
%files -n python3-libpamtest
%{_libdir}/python3.7/site-packages/pypamtest.so
```

Reply:
I would prefer that we use the following line at the top of the spec to define `%python3_sitearch` when it is not already defined. This makes the spec file port more easily to our 2.0 branch, which does have that macro defined.

```
%define python3_sitearch %(python3 -c "from distutils.sysconfig import get_python_lib; import sys; sys.stdout.write(get_python_lib(1))")
```

Reply:
got it